File: examples/chapter08/e01_mean_squared_error/python/main.py
import torch
import torch.nn.functional as F

pred = torch.tensor([[1., 2.], [3., 4.]])
target = torch.tensor([[2., 1.], [3., 2.]])

loss = F.mse_loss(pred, target, reduction="mean")

# squared errors: [1, 1, 0, 4] -> mean = 1.5
print("MSE =", loss.item())  # MSE = 1.5

File: examples/chapter08/e01_mean_squared_error/rust/src/main.rs
use burn::backend::NdArray;
use burn::nn::loss::{MseLoss, Reduction};
use burn::tensor::Tensor;

type B = NdArray;

fn main() {
    let dev = Default::default();

    let pred = Tensor::<B, 2>::from_floats([[1.0, 2.0], [3.0, 4.0]], &dev);
    let target = Tensor::<B, 2>::from_floats([[2.0, 1.0], [3.0, 2.0]], &dev);

    let mse = MseLoss::new();
    let loss = mse.forward(pred.clone(), target.clone(), Reduction::Mean);

    println!("MSE = {}", loss.to_data()); // mean of [1, 1, 0, 4] = 1.5
}
